mewarpx/mewarpx/diags_store/diag_base.py

```diff
@@ -194,21 +196,29 @@ def __init__(self, diag_steps, preset_string='default',
         if install:
             callbacks.installafterstep(self.text_diag)
 
+        self.particle_steps_total = 0
+        self.previous_particle_steps_total = 0
+
     def text_diag(self):
         """Write requested information to output."""
+
         if self.check_timestep():
             live_parts, parts_per_species_str = self._get_part_nums()
-
+            # This isn't perfectly accurate, but it's a good approximation
+            self.particle_steps_total += live_parts * self.diag_steps
             # Loop everything else so run doesn't crash on diag error
             try:
                 wall_time = time.time() - self.prev_time
                 steps = mwxrun.get_it() - self.prev_step
```

Comment:
Just one last thing. I believe the diagnostics are supposed to be able to handle irregular diagnostic intervals so we should probably use `steps` rather than `self.diag_steps` in the `self.particle_steps_total` update.

Reply:
Ah ok